bcgov/biohubbc
# BHBC-2267: Create Survey context

# Overview

Creates a Survey Context, which wraps survey-related app sections, providing a centralized location for fetching/refreshing survey data.

## Links to Jira tickets

 - https://quartech.atlassian.net/browse/BHBC-2267

## Description of relevant changes
 - Creates the `useSurveyContext` hook
 - Wraps all survey routes in the `ProjectRouter` with the `SurveyContextProvider`
 - Removes any component props that pass a Survey object

## Testing
 1. Create a project if you don't have one
 2. Create a survey, if you dont' have one
 3. Click on the survey
 - [x] Ensure that the survey loads
 4. Upload an observations
 - [x] Ensure that the observation loads
 5. Upload a report and attachment
 - [x] Ensure that attachments load
